Whiskey/Functions/Write-WhiskeyInfo.ps1

```diff
@@ -1,55 +1,153 @@
 
     .EXAMPLE
-    Write-WhiskeyInfo -Context $context -Message 'An info message' -Indent 2
+    Write-WhiskeyInfo -Context $context -Message 'An info message'
 
-    Demonstrates how write an `Info` message with a custom indent.
+    Demonstrates how write an `Info` message.
     #>
     [CmdletBinding()]
     param(
-        [Parameter(Mandatory)]
-        # The current context.
+        # The context for the current build. If not provided, Whiskey will search up the call stack looking for it.
         [Whiskey.Context]$Context,
 
-        [Parameter(Mandatory,ValueFromPipeline)]
-        [AllowEmptyString()]
-        [AllowNull()]
-        # The message to write.
-        [String]$Message,
+        [ValidateSet('Error','Warning','Info','Verbose','Debug')]
+        # The log level.
+        [String]$Level = 'Info',
 
-        [int]$Indent = 0
+        [Parameter(Mandatory,ValueFromPipeline,Position=0)]
+        [AllowNull()]
+        [AllowEmptyString()]
+        # The message to log.
+        [String[]]$Message
     )
 
-    process
+    begin
     {
         Set-StrictMode -Version 'Latest'
         Use-CallerPreference -Cmdlet $PSCmdlet -SessionState $ExecutionContext.SessionState
 
-        $Message = Format-WhiskeyMessage @PSBoundParameters
-        if( $supportsWriteInformation )
+        # Only write if absolutely necessary as it can be expensive.
+        # Since silent errors, warnings, and info can be captured, we still have to output when their prefs are silent.
+        # Silent verbose and debug messages are currently capturable, so don't even bother to write them.
+        $write = ($Level -eq 'Error' -and $ErrorActionPreference -ne [Management.Automation.ActionPreference]::Ignore) -or 
+                 ($Level -eq 'Warning' -and $WarningPreference -ne [Management.Automation.ActionPreference]::Ignore) -or
+                 ($Level -eq 'Info' -and $InformationPreference -ne [Management.Automation.ActionPreference]::Ignore) -or
+                 ($Level -eq 'Verbose' -and $VerbosePreference -notin @([Management.Automation.ActionPreference]::Ignore,[Management.Automation.ActionPreference]::SilentlyContinue)) -or
+                 ($Level -eq 'Debug' -and $DebugPreference -notin @([Management.Automation.ActionPreference]::Ignore,[Management.Automation.ActionPreference]::SilentlyContinue))
+
+        if( -not $write )
+        {
+            return
+        }
+
+        if( -not $context )
+        {
+            $context = Get-WhiskeyContext
+        }
+
+        $messages = New-Object 'Collections.ArrayList'
+
+        $task = ''
+        if( -not $context )
+        {
+            $duration = (Get-Date).ToString('HH:mm:ss.ff')
+        }
+        else
+        {
+            $duration = ((Get-Date) - $context.StartedAt).ToString('hh":"mm":"ss"."ff')

```

Comment:
This is on purpose. The first case is showing the time of day because we couldn't find a context so we don't know how long it has been since the build started. That's a date/time format string not a time span format string.